x/gamm/pool-models/stableswap/pool.go

```diff
@@ -20,13 +20,20 @@ var _ types.PoolI = &Pool{}
 // * len(initialLiquidity) = 2
 // * FutureGovernor is valid
 // * poolID doesn't already exist
-func NewStableswapPool(poolId uint64, stableswapPoolParams PoolParams, initialLiquidity sdk.Coins, futureGovernor string) (Pool, error) {
+func NewStableswapPool(poolId uint64, stableswapPoolParams PoolParams, initialLiquidity sdk.Coins, scalingFactors []uint64, futureGovernor string) (Pool, error) {
+	if len(scalingFactors) == 0 {
+		scalingFactors = []uint64{1, 1}
+	} else if scalingFactors[0] == 0 || scalingFactors[1] == 0 {
+		scalingFactors = []uint64{1, 1}
```

Comment:
```suggestion
		return Pooll{}, errors.New("one of the scaling factors provided was 0")
```

Reply:
(reverts back to your prior commit, I think that was right)

Reply:
If this was on the osmosis main repo, I'd just commit this suggestion and then hit merge. After this is updated, I think its good to merge.